splash/network_manager.py

```diff
@@ -123,6 +123,38 @@ def createRequest(self, operation, request, outgoingData=None):
             self.log(traceback.format_exc(), min_level=1, format_msg=False)
             return super(ProxiedQNetworkAccessManager, self).createRequest(operation, request, outgoingData)
 
+    def _get_request_body(self, request, outgoing_data):
+        if outgoing_data is not None:
```

Comment:
It is often easier to read functions when special cases are handled first:

```py
if outgoing_data is None:
    return None
# ... the rest
```
this also provides more horizontal space (less nesting).